_posts/2021-02-19-26-rln-relay.md

```diff
@@ -0,0 +1,195 @@
+In this solution, privacy is guaranteed since: 1) Peers do not have to disclose any piece of personal information in any phase i.e., neither in the registration nor in the messaging phase 2) Peers can prove that they have not exceeded the messaging rate in a zero-knowledge manner and without leaving any trace to their membership accounts. 
+Furthermore, all the computations are light hence this solution fits the heterogenous p2p messaging system. Note that  the zero-knowledge proof parts are handled through zkSNARKs  and  the benchmarking result can be found in RLN benchmark [^5].
+
+**Future steps**: We are still at the PoC level, and the development is in progress. As our future steps, 
+- we would like to evaluate the running time associated with the Merkle tree operations. Indeed, the need to locally store Merkle tree on each peer was one of the unknowns discovered during this PoC and yet the concrete benchmarking result in this regard is not available. 
+- We would also like to persue our storage-efficient Merkle Tree maintenance solution in order to lower the storage overhead of peers. 
+- In line with the storage optimization, the full-node light-node structure is another path to follow. However, there is already a privacy concern about the full-light node approach where light nodes have to expose their index in the tree to the full-nodes in order to fetch their membership proof /authentication path. The peer's index is somewhat a private data hence the privacy implication of such exposure must be studied and analyzed as well. 
+- Another possible improvement is to replace the membership contract with a distributed group management scheme e.g., through distributed hash tables. This is to address possible performance issues that the interaction with the Ethereum blockchain may cause.  For example, the registration transactions are subject to delay as they have to be mined before being visible in the state of the membership contract. This means peers have to wait for some time before being able to publish any message.
+
+
+
+# Acknowledgement
+Thanks to Onur Kılıç for his explanation and pointers and for helping with development and runtime issues. Also thanks to Barry Whitehat for his time and insightful comments.
+
+# References
+
+
+[^1]: Waku v2: https://github.com/vacp2p/specs/blob/master/specs/waku/v2/waku-v2.md
+[^2]: RLN-Relay specifications: https://github.com/vacp2p/vacp2p.github.io/blob/RLN-Relay/_posts/2021-02-19-26-RLN-Relay.md
+[^3]: RLN documentation: https://hackmd.io/tMTLMYmTR5eynw2lwK9n1w?both
+[^4]: RLN repository: https://github.com/kilic/RLN
+[^5]: RLN Benchmark: https://hackmd.io/tMTLMYmTR5eynw2lwK9n1w?view#Benchmarks
+[^6]: Peer Scoring: https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#peer-scoring
+[^7]: Peer scoring security issues: https://github.com/vacp2p/research/issues/44
+[^8]: Proof of work: https://www.investopedia.com/terms/p/proof-work.asp
```

Comment:
Hm, not the best reference probably :sweat_smile: Maybe Bitcoin paper? And/Or Waku spec? There's a great ref re PoW for spam protection from 80s in anon comm literature, probably in Zotero, but can't find it right now